File: Getting Started - WCF RIA Services/[VB]-Getting Started - WCF RIA Services/VB/HRApp/Helpers/TargetNullValueConverter.vb
Imports System
Imports System.Windows.Data

''' <summary>
'''     Two way IValueConverter that lets you bind a property on a bindable object
'''     to a dependency property when the dependency property treats empty as null
'''     and the object property treats empty as the empty string
''' </summary>
Public Class TargetNullValueConverter
    Implements IValueConverter
    
    Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.Convert
        Return If(String.IsNullOrEmpty(DirectCast(value, String)), Nothing, value)
    End Function

    Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As System.Globalization.CultureInfo) As Object Implements IValueConverter.ConvertBack
        Return If(value Is Nothing, String.Empty, value)
    End Function
End Class

File: Getting Started - WCF RIA Services/[C#]-Getting Started - WCF RIA Services/C#/HRApp/Helpers/TargetNullValueConverter.cs
namespace HRApp
{
    using System;
    using System.Windows.Data;

    /// <summary>
    ///     Two way IValueConverter that lets you bind a property on a bindable object
    ///     that can be an empty string value to a dependency property that should 
    ///     be set to null in that case
    /// </summary>
    public class TargetNullValueConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return string.IsNullOrEmpty((string)value) ? null : value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
        {
            return value ?? string.Empty;
        }
    }
}
